src/socks.rs

```diff
@@ -181,12 +200,24 @@ impl RpcCachedSock {
     /// messages are only popped upon successful send.
     pub fn flush_out_fast(&mut self) {
         while let Some((msg, peer)) = self.cache.read_head() {
-            if msg.send(&self.sock, peer).is_ok() {
-                self.cache.pop_front();
+            if let Err(e) = self.unix_send(msg, peer) {
+                if e.kind() != ErrorKind::WouldBlock {
+                    error!("Failure sending over unix sock: {}", e);
+                }
+                return;
             } else {
-                break;
+                /* drop it, we sent it already */
+                self.cache.pop_front();
             }
         }
+        debug_assert!(self.cache.is_empty());
+        if self.interests().is_writable() {
+            trace!("Writable readiness notification no longer needed");
+            self.interests = Interest::READABLE;
+            //            let _ = self.interests.remove(Interest::WRITABLE);
+            //            assert!(!self.interests.is_writable());
+            //            assert!(self.interests.is_readable());
```

Comment:
Leftovers?

Reply:
Yeah, to investigate later. I got a behavior that smelt like a bug.